pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -646,6 +646,11 @@ func (c *Controller) syncRule(key string) error {
 		return nil
 	}
 
+	if !c.antreaProxyEnabled && rule.Services != nil {
+		klog.V(2).InfoS("Skipping rule with `Services` field when AntreaProxy is not enabled", "ruleID", key)
+		return nil
+	}
```

Comment:
Looks like this should be in a separate PR?

cc @tnqn

Reply:
This doesn't seem correct to me. Why would L4 matching rely on AntreaProxy?

Reply:
Thanks for reminding, the field should be `rule.To.ToService`, not `rule.Services`.